azkaban-exec-server/src/main/java/azkaban/execapp/JobRunner.java

```diff
@@ -210,38 +217,118 @@ private void createLogger() {
-        fileAppender.setMaxFileSize(jobLogChunkSize);
-        jobAppender = fileAppender;
-        logger.addAppender(jobAppender);
-        logger.setAdditivity(false);
+        attachFileAppender(createFileAppender());
       } catch (IOException e) {
+        removeAppender(jobAppender);
         flowLogger.error("Could not open log file in " + workingDir
             + " for job " + this.jobId, e);
       }
+
+      AzkabanExecutorServer exec = AzkabanExecutorServer.getApp();
+      if (exec != null && !exec.getAzkabanProps().getBoolean("kafkaLogging.globalDisable", false)
+          && props.getBoolean("azkaban.job.kafkaLogging.enabled", false)) {
+        try {
+          attachKafkaAppender(createKafkaAppender());
+        } catch (Exception e) {
+          removeAppender(kafkaAppender);
+          flowLogger.error("Failed to create Kafka appender for job " + this.jobId, e);
+        }
+      }
     }
   }
 
+  private void attachFileAppender(FileAppender appender) {
+    // If present, remove the existing file appender
+    if (jobAppender != null) {
+      removeAppender(jobAppender);
+      flowLogger.info("Removed old file appender for job " + this.jobId);
+    }
+
+    jobAppender = appender;
+    logger.addAppender(jobAppender);
+    logger.setAdditivity(false);
+    flowLogger.info("Attached file appender for job " + this.jobId);
+  }
+
+  private FileAppender createFileAppender() throws IOException {
+    // Set up log files
+    String logName = createLogFileName(node);
+    logFile = new File(workingDir, logName);
+    String absolutePath = logFile.getAbsolutePath();
+
+    // Attempt to create FileAppender
+    RollingFileAppender fileAppender =
+        new RollingFileAppender(loggerLayout, absolutePath, true);
+    fileAppender.setMaxBackupIndex(jobLogBackupIndex);
+    fileAppender.setMaxFileSize(jobLogChunkSize);
+
+    flowLogger.info("Created file appender for job " + this.jobId);
+    return fileAppender;
+  }
+
   private void createAttachmentFile() {
     String fileName = createAttachmentFileName(node);
     File file = new File(workingDir, fileName);
     attachmentFileName = file.getAbsolutePath();
   }
 
+  private void attachKafkaAppender(KafkaLog4jAppender appender) {
+    // If present, remove the existing kafka appender
+    if (kafkaAppender.isPresent()) {
+      removeAppender(kafkaAppender);
+      flowLogger.info("Removed old Kafka appender for job " + this.jobId);
+    }
+
+    kafkaAppender = Optional.of(appender);
+    logger.addAppender(kafkaAppender.get());
+    logger.setAdditivity(false);
+    flowLogger.info("Attached new Kafka appender for job " + this.jobId);
+  }
+
+  private KafkaLog4jAppender createKafkaAppender() {
+    KafkaLog4jAppender kafkaProducer = new KafkaLog4jAppender();
+    Props azkProps = AzkabanExecutorServer.getApp().getAzkabanProps();
+    kafkaProducer.setSyncSend(false);
+    kafkaProducer.setBrokerList(azkProps.getString("kafkaLogging.brokerList"));
```

Comment:
Use constants for config names and follow a consistent naming convention as suggested above?